small/lsregion_asan.c

```diff
@@ -76,7 +107,7 @@ lsregion_to_iovec(const struct lsregion *lsregion, struct iovec *iov,
 		if (cnt >= max_iovcnt)
 			break;
 		iov->iov_base = small_asan_payload_from_header(alloc);
-		iov->iov_len = alloc->size;
+		iov->iov_len = alloc->used;
 		svp->slab_id = alloc->id;
```

Comment:
Let's also add an assertion `lsregion_to_iovec` is not used after reservation but without allocation. Looks like it will return  incorrect result.

Reply:
AFAICS there are no problems here.
`svp->slab_id` is initialized with `LSLAB_NOT_USED_ID`, and the same id is set for the unused allocations:
https://github.com/tarantool/small/blob/c6a3917501134d634316a1c03cbd094ced0f8aef/include/small/lsregion.h#L54-L59
https://github.com/tarantool/small/blob/a5139c23f73c67d12847355831e61615351e7a4b/small/lsregion_asan.c#L58
so here an unused allocation will be simply skipped by condition `alloc->id <= prev_id`:
https://github.com/tarantool/small/blob/c6a3917501134d634316a1c03cbd094ced0f8aef/small/lsregion_asan.c#L68-L75